src/main/java/shop/dao/impl/OrderDAOImpl.java

```diff
@@ -0,0 +1,52 @@
+package shop.dao.impl;
+
+import shop.dao.OrderDAO;
+import shop.dao.UserDAO;
+import shop.entity.Order;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class OrderDAOImpl implements OrderDAO {
+
+    // this is your main place where all orders will be saved
+    private Map<String, Order> orderMap = new HashMap<>();
+    private UserDAO userDAO = new UserDAOImpl();
+
+    @Override
+    public void save(Order order) {
+        orderMap.put(order.getUserId(), order);
```

Comment:
before putting to map please generate ID for Order using UUID (you can see it in UserDAOImpl